pubsub/google/cloud/pubsub/topic.py

```diff
@@ -426,11 +434,16 @@ class Batch(object):
                          before automatically commiting. Defaults to infinity
                          (off).
     :type max_messages: float
+
+    :param max_size: The maximum size that the serialized messages can be
+                     before automatically commiting. Defaults to just under
+                     10 MB.
+    :type max_size: int
     """
     _INFINITY = float('inf')
 
     def __init__(self, topic, client, max_interval=_INFINITY,
-                 max_messages=_INFINITY):
+                 max_messages=_INFINITY, max_size=10e6):
```

Comment:
If the published limit is 10 MB, I wonder if it might make more sense to have the batching limit be a little smaller (e.g. 9 MB). There is nothing here that errors out if you send a single message above `max_size`, and for the common use case (sending a bunch of messages), having auto-commit tack in at 90% of the limit is probably actually the most useful behavior.